### DISCUSSION TITLE
contextualSpellCheck: Misbehavior?

### DISCUSSION THREAD

**Discussion by Halvani** [explosion/spaCy#10604] [Help: Model Advice] [answered]
```
import spacy
import contextualSpellCheck

nlp = spacy.load('en_core_web_sm')
contextualSpellCheck.add_to_pipe(nlp)

doc = nlp('Conference on Empirical Methodsin Natural Language Processing')
print(doc._.outcome_spellCheck)

# Conference on Empirical Analysis Natural Language Processing
```
I'm not sure if a spell checker should actually perform a paraphrase. 
Shouldn't it just correct the spelling errors? ...so that the correct output becomes:

"**Conference on Empirical Methods in Natural Language Processing**"

**Comment by ljvmiranda921** [comment, marked as answer]
Hi @Halvani , this might be due to the fact that `contextualSpellChecker` uses BERT embeddings (i.e., checking the context of a given token) to estimate how a word should be corrected. You may have a shot by preprocessing the text a bit to better guide the spell-checker, trying out a rules-based approach (perhaps by using other tools like [autocorrect](https://github.com/filyp/autocorrect)), or experimenting with [some of the spell checker's parameters](https://github.com/R1j1t/contextualSpellCheck/blob/master/contextualSpellCheck/contextualSpellCheck.py#L25). For the last suggestion, perhaps you can try playing around the edit distance or the BERT model? Just to see how strict it will be when correcting your text.
